### ISSUE TITLE
Corrupt Excel Files after changing column names

### ISSUE THREAD

**Issue by LimitlessWorkaround** [abap2xlsx/abap2xlsx#554]
I wrote a report which exports data fom a table and creates a excel file which is saved on the pc afterwards. The problem is that the files are corrupt but only when I change the column names in my program.

I have some structs like this:

```
types:
      begin of struct_example,
        name(25)      type c,
        job(15)          type c,
        id(15)            type c,
      end of struct_example.
```
When I export the data without any changes then the column names look like this: "Col: NAME"...
I change them dynamically according to the users language. E.g. from "Col: NAME" to "Name".

```
do l_no_of_cols times.                                                (Do "number of columns" times)

          try.
              lr_worksheet->GET_CELL(                            (Get data from cell)
                EXPORTING
                  IP_COLUMN  = lv_count
                  IP_ROW     = 1
                  IMPORTING
                    EP_VALUE   = lv_value
                ).
            catch zcx_excel .
          endtry.
          if lv_value(3) = 'Col'.                                      (If column name starts with "Col" then change it...)
            lv_value = lv_value+5. 
            lv_tmp = lv_value+1.
            translate lv_tmp to lower case.
            lv_value = |{ lv_value(1) }{ lv_tmp }|.
            try.
                lr_worksheet->SET_CELL(                       (Set new column name)
                EXPORTING
                  IP_COLUMN    = lv_count
                  IP_ROW       = 1
                  IP_VALUE     = lv_value
                ).
              catch zcx_excel .
            endtry.

```

![image](https://user-images.githubusercontent.com/21292736/43896594-9d43dd5a-9bd9-11e8-86d3-def89457de6e.png)

Basically it says "There are some errors with the data in "Example.xlsx". Should we restore as much as possible? If you trust this worksheet the proceed with "Yes""

After proceeding the file opens correctly and then there is this warning:

![image](https://user-images.githubusercontent.com/21292736/43896746-260bc5c6-9bda-11e8-9a6c-88143aacfb74.png)

Which can be translated to: "repaired data: table of /xl/tables/table1.xml-part (table)"

The linked log on the bottom shows the following: 

After opening the file I get the following errors:

![image](https://user-images.githubusercontent.com/21292736/43896975-dde146c6-9bda-11e8-96ab-96f07ccd107d.png)

Does someone know how to solve the problem?

**Comment by gregorwolf** [comment]
Do you create the table completely in abap2xlsx or do you use a template?

**Comment by LimitlessWorkaround** [comment]
@gregorwolf I'm very new to abap and abap2xlsx. I'll just post my code here...

Creating the xlsx file:

```
method create_xlsx.

    DATA: lr_excel     TYPE REF TO zcl_excel,
          lr_worksheet TYPE REF TO zcl_excel_worksheet.
    DATA: ls_table_settings TYPE zexcel_s_table_settings.
    DATA: lr_writer    TYPE REF TO zif_excel_writer.
    DATA: l_xstring    TYPE xstring.
    DATA: l_no_of_cols TYPE i,
          l_column     TYPE zexcel_cell_column_alpha,
          lv_count     type i,
          lv_value     type string,
          lv_tmp       type string.
    DATA: lr_tabledescr TYPE REF TO cl_abap_tabledescr,
          lr_strucdescr TYPE REF TO cl_abap_structdescr.

    CREATE OBJECT lr_excel.

    lr_worksheet = lr_excel->get_active_worksheet( ).

    ls_table_settings-table_style      = zcl_excel_table=>builtinstyle_medium2.
    ls_table_settings-show_row_stripes = abap_true.
    ls_table_settings-nofilters        = abap_false.
    ls_table_settings-top_left_column  = 'A'.
    ls_table_settings-top_left_row     = '1'.
    ls_table_settings-table_name       = title.
    TRY.
        lr_worksheet->bind_table( ip_table          = lt_rollen
                                  is_table_settings = ls_table_settings ).
        lr_worksheet->freeze_panes( ip_num_rows = 1 ).
      CATCH zcx_excel.
    ENDTRY.

    TRY.
        lr_tabledescr ?= cl_abap_tabledescr=>describe_by_data( lt_rollen ).
        lr_strucdescr ?= lr_tabledescr->get_table_line_type( ).
        l_no_of_cols = lines( lr_strucdescr->components ).
      CATCH cx_root.
    ENDTRY.

*column name changing function*

DO l_no_of_cols TIMES.
      TRY.
          CALL METHOD zcl_excel_common=>convert_column2alpha
            EXPORTING
              ip_column = sy-index
            RECEIVING
              ep_column = l_column.
        CATCH zcx_excel .
          EXIT.
      ENDTRY.
      TRY.
          lr_worksheet->get_column_dimension( l_column )->set_auto_size( ip_auto_size = abap_true ).
        CATCH zcx_excel.
      ENDTRY.
    ENDDO.

    CREATE OBJECT lr_writer TYPE zcl_excel_writer_2007.

    l_xstring = lr_writer->write_file( lr_excel ).

    lv_bytecount = xstrlen( l_xstring ).
    lv_rawdata = cl_bcs_convert=>xstring_to_solix( iv_xstring  = l_xstring ).

endmethod.
```

saving the file to my pc:

```
method download.
    Data: filename Type string.

    IF filename CA '/'.
      REPLACE REGEX '([^/])\s*$' IN filename WITH '$1/' .
    ELSE.
      REPLACE REGEX '([^\\])\s*$' IN filename WITH '$1\\'.
    ENDIF.

    filename = |{ p_path }\\{ title }.xlsx|.

    cl_gui_frontend_services=>GUI_DOWNLOAD(
      EXPORTING bin_filesize          = lv_bytecount
                filename              = filename
                filetype              = 'BIN'
                CONFIRM_OVERWRITE     = 'X'
    CHANGING data_tab              = lv_rawdata
    EXCEPTIONS
     .....
    ).
    IF SY-SUBRC <> 0.
      MESSAGE ID SY-MSGID TYPE SY-MSGTY NUMBER SY-MSGNO
                 WITH SY-MSGV1 SY-MSGV2 SY-MSGV3 SY-MSGV4.
    ENDIF.
  endmethod.
```

**Comment by gregorwolf** [comment]
Have you checked out the demo report ZDEMO_EXCEL3? When you run it and you're logged in in German, the headings are taken from the ABAP Dictionary and sown in German. When you log on in English you get them in English:

![screen shot 2018-08-09 at 23 50 21](https://user-images.githubusercontent.com/2639535/43927826-1d55e894-9c2f-11e8-977c-110cd0712358.png)

Isn't that the better solution?

**Comment by LimitlessWorkaround** [comment]
The problem is that I'm not using any Dictionary data types. Most of the fields are simple chars.

**Comment by gregorwolf** [comment]
What's about using the IT_FIELD_CATALOG parameter of BIND_TABLE?

**Comment by LimitlessWorkaround** [comment]
Thank you very much. It's working now
